pkg/cache/multi_namespace_cache.go

```diff
@@ -35,8 +35,11 @@ import (
 type NewCacheFunc func(config *rest.Config, opts Options) (Cache, error)
 
 // MultiNamespacedCacheBuilder - Builder function to create a new multi-namespaced cache.
-// This will scope the cache to a list of namespaces. Listing for all namespaces
-// will list for all the namespaces that this knows about.
+// This will scope the cache to a list of namespaces. Note that its usage is NOT recommended
```

Comment:
I think we should keep the `Listing for all namespaces will list for all the namespaces that this cache knows about. ` sentence.

Maybe `Note that this is not intended to be used for excluding namespaces, this is better done via a Predicate. Also note that you may face performance issues when using this with a high number of namespaces.`  is as a more precise description?

And regarding the performance issue, is that much of a difference compared to using a global cache?

Reply:
Applied suggestions.